src/python/bot/fuzzers/syzkaller/fuzzer.py

```diff
@@ -0,0 +1,167 @@
+    """Inits the SyzkallerRunner.
+
+    Args:
+      executable_path: Path to the fuzzer executable.
+      default_args: Default arguments to always pass to the fuzzer.
+    """
+    super(SyzkallerRunner, self).__init__(
+        executable_path=executable_path, default_args=None)
+
+  def get_command(self, additional_args=None):
+    """Process.get_command override."""
+    base_command = super(SyzkallerRunner,
+                         self).get_command(additional_args=additional_args)
+
+    return base_command
+
+  def _create_empty_testcase_file(self):
+    """Create an empty testcase file in temporary directory."""
+    _, path = tempfile.mkstemp(dir=fuzzer_utils.get_temp_dir())
+    return path
+
+  def get_testcase_path(self, log_lines):
+    """Get testcase path from log lines."""
+    for line in log_lines:
+      match = re.match(KASAN_CRASH_TESTCASE_REGEX, line)
+      if match:
+        return match.group(1)
+
+    return None
+
+  def fix_timeout_argument_for_reproduction(self, arguments):
+    """Changes timeout argument for reproduction. This is higher than default to
+    avoid noise with smaller fuzzing defaults."""
+    fuzzer_utils.extract_argument(arguments, constants.TIMEOUT_FLAG)
+    arguments.append(
+        '%s%d' % (constants.TIMEOUT_FLAG, constants.REPRODUCTION_TIMEOUT_LIMIT))
+
+  def fuzz(self,
+           corpus_directories,
+           fuzz_timeout,
+           artifact_prefix=None,
+           additional_args=None,
+           extra_env=None):
+    """This is where actual syzkaller fuzzing is done."""
+    del corpus_directories, artifact_prefix, extra_env
+    additional_args = copy.copy(additional_args)
+    fuzz_result = self.run_and_wait(
+        additional_args=additional_args, timeout=fuzz_timeout)
+
+    log_lines = utils.decode_to_unicode(fuzz_result.output).splitlines()
+    fuzz_result.output = None
+    crash_testcase_file_path = self.get_testcase_path(log_lines)
+
+    if not crash_testcase_file_path and fuzz_result.return_code:
+      crash_testcase_file_path = self._create_empty_testcase_file()
+
+    fuzz_logs = '\n'.join(log_lines)
+
+    # TODO: Parse stats information and add them to FuzzResult
+    parsed_stats = []
+
+    crashes = []
+    if crash_testcase_file_path:
+      arguments = additional_args[:]
+      #libfuzzer.remove_fuzzing_arguments(arguments)
+      arguments = additional_args[:]
+      reproduce_arguments = arguments[:]
+      self.fix_timeout_argument_for_reproduction(reproduce_arguments)
+      actual_duration = int(fuzz_result.time_executed)
+      # Write the new testcase.
+      # Copy crash testcase contents into the main testcase path.
+      crashes.append(
+          Crash(crash_testcase_file_path, fuzz_logs, reproduce_arguments,
+                actual_duration))
+
+    return FuzzResult(fuzz_logs, fuzz_result.command, crashes, parsed_stats,
+                      fuzz_result.time_executed)
+
+
+class Crash(object):
```

Comment:
Remove these and return the base engine.Crash/FuzzResult objects.

Reply:
Done